### DISCUSSION TITLE
v1.72.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#3574] [Announcements]
# v1.72.0

## Breaking Changes ⚠️

- The mobile app needs to be on the same version of the server v1.72.0 to operate correctly.
- Drop support for armV7

## Highlights

Welcome to release `v1.72.0` of Immich. This release includes improvements across the board, from UI elements on both the web and the mobile app client to the server’s support for hardware transcoding. Some of the key highlights include:

- Initial support for hardware transcoding.
- Manually upload assets from the mobile app.
- Optimized views on the web.
- Album description.

### Hardware transcoding

Thanks, @mertalev, for working on this feature. You can now use a GPU to speed up video transcoding in Immich. We hope to receive feedback from you on this experimental feature.

The `docker-compose.yml` file is updated to support this feature. You can read about how to enable hardware transcoding [here](https://immich.app/docs/features/hardware-transcoding) 

### Manually upload assets from the mobile app

You can now choose single or multiple photos and videos to perform selective backup in addition to the existing mass backup mechanism.

<p align="center">
<img src="https://github.com/immich-app/immich/assets/139912620/00e1a1ce-0d6d-4be7-a5e4-c3f50781c0d0 " alt="Upload Asset Example" height="300">
<img src="https://github.com/immich-app/immich/assets/139912620/b0d7efec-a77f-4474-9e29-c0695d7c3918" alt="Upload Asset from Timeline Example" height="300">
</p>

### Optimized views on the web

This refactor has been in the backlog for a while, and thanks to @jrasm91 for working on this. The optimized virtual viewport is now used in the following views on the web.

- Main timeline
- Archive
- Favorite
- Person’s detail

We plan to add this virtual viewport to all the views in the future to ensure a top-notch browsing experience.

### Album description

Album now has a description field; you can add/edit it on the web. The description can also be used to search for an album using `m:<album-description>` syntax. Mobile doesn't have this feature yet.

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Mobile
* fix(mobile): clip and change background for asset selection by @shalong-tanwen in https://github.com/immich-app/immich/pull/3460
* fix(mobile): fix asset order in album from latest to oldest by @alextran1502 in https://github.com/immich-app/immich/pull/3469
* feat(mobile): auto close memories on scrolling beyond first / last memory by @shalong-tanwen in https://github.com/immich-app/immich/pull/3476
* fix(mobile): Improve vertical swipe detection in gallery viewer by @mPyKen in https://github.com/immich-app/immich/pull/3486
* fix(mobile): "preselectedAssets" parameter not working by @xiangzy1 in https://github.com/immich-app/immich/pull/3516
* fix(mobile): local merged videos should now play locally instead of loading remote by @martyfuhry in https://github.com/immich-app/immich/pull/3521
* feat(mobile): Manual asset upload by @shalong-tanwen in https://github.com/immich-app/immich/pull/3445

### Web
* refactor(web): asset grid stores by @brighteyed in https://github.com/immich-app/immich/pull/3464
* fix(web): minor issues by @brighteyed in https://github.com/immich-app/immich/pull/3499
* refactor(web): asset grid state by @jrasm91 in https://github.com/immich-app/immich/pull/3513
* fix(web): leave assets selected after creating a shared link by @jrasm91 in https://github.com/immich-app/immich/pull/3531
* feat(web): :sparkles: Rounded corners on selected assets by @paulthvt in https://github.com/immich-app/immich/pull/3524
* refactor(web): asset store by @jrasm91 in https://github.com/immich-app/immich/pull/3528
* fix(web): scrollbar by @jrasm91 in https://github.com/immich-app/immich/pull/3536
* feat(web): handle escape key and higher wheel zoom ratio by @martabal in https://github.com/immich-app/immich/pull/3471
* fix(web): scrollbar by @jrasm91 in https://github.com/immich-app/immich/pull/3536
* chore(web): organize prettier trigger order by @alextran1502 in https://github.com/immich-app/immich/pull/3538
* refactor(web): move memory lane to photos page by @jrasm91 in https://github.com/immich-app/immich/pull/3541
* refactor(server,web): time buckets for main timeline, archived, and favorites (1) by @jrasm91 in https://github.com/immich-app/immich/pull/3537
* feat(web): use time buckets of person detail page (3) by @jrasm91 in https://github.com/immich-app/immich/pull/3557
* fix(web): asset selection show up on top by @alextran1502 in https://github.com/immich-app/immich/pull/3563
* fix(web): broken search-bar during page load by @martabal in https://github.com/immich-app/immich/pull/3548
* feat(server/web): album description by @jrasm91 in https://github.com/immich-app/immich/pull/3558
* refactor(web): time buckets (2) by @jrasm91 in https://github.com/immich-app/immich/pull/3554

### Machine Learning
* refactor(ml): model downloading by @mertalev in https://github.com/immich-app/immich/pull/3545

### Server
* feat(web/server): Search by panorama photos by @mPyKen in https://github.com/immich-app/immich/pull/3470
* refactor(server): test fixtures by @jrasm91 in https://github.com/immich-app/immich/pull/3491
* test(server): auth e2e by @jrasm91 in https://github.com/immich-app/immich/pull/3492
* chore(server): auto sort open api spec by @jrasm91 in https://github.com/immich-app/immich/pull/3500
* fix(server): apply timezone to storage template by @jrasm91 in https://github.com/immich-app/immich/pull/3504
* refactor(server)!: add/remove album assets by @jrasm91 in https://github.com/immich-app/immich/pull/3109
* feat(server): transcoding hardware acceleration by @mertalev in https://github.com/immich-app/immich/pull/3171
* fix(server): get remote ip by @jrasm91 in https://github.com/immich-app/immich/pull/3515
* chore(server): process title by @jrasm91 in https://github.com/immich-app/immich/pull/3519
* fix(server): ffprobe not found by @jrasm91 in https://github.com/immich-app/immich/pull/3532
* refactor(server): standardize user controller by @jrasm91 in https://github.com/immich-app/immich/pull/3501
* fix(admin-cli): load env files by @jrasm91 in https://github.com/immich-app/immich/pull/3503
* refactor(server): standardize user controller by @jrasm91 in https://github.com/immich-app/immich/pull/3501

### Documentation and Others
* docs: Clarify immich-admin command usage by @mnahinkhan in https://github.com/immich-app/immich/pull/3456
* docs: Include information on GHCR login for docker installation by @mnahinkhan in https://github.com/immich-app/immich/pull/3457
* docs: fix the grammar for albums to be backed up by @Jacob-Roberts in https://github.com/immich-app/immich/pull/3461
* blog: july 2023 by @alextran1502 in https://github.com/immich-app/immich/pull/3466
* docs: Fix typo in logo docs by @mladedav in https://github.com/immich-app/immich/pull/3475
* docs: update architecture.md by @jrasm91 in https://github.com/immich-app/immich/pull/3488
* docs: Add Spanish translations of Readme by @jorge-dev in https://github.com/immich-app/immich/pull/3511
* chore: drop support armV7 by @alextran1502 in https://github.com/immich-app/immich/pull/3509
* chore: add Alex mac mini as runner by @alextran1502 in https://github.com/immich-app/immich/pull/3526

## New Contributors
* @mnahinkhan made their first contribution in https://github.com/immich-app/immich/pull/3456
* @Jacob-Roberts made their first contribution in https://github.com/immich-app/immich/pull/3461
* @mladedav made their first contribution in https://github.com/immich-app/immich/pull/3475
* @mPyKen made their first contribution in https://github.com/immich-app/immich/pull/3470
* @jorge-dev made their first contribution in https://github.com/immich-app/immich/pull/3511
* @paulthvt made their first contribution in https://github.com/immich-app/immich/pull/3524
* @xiangzy1 made their first contribution in https://github.com/immich-app/immich/pull/3516

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.71.0...v1.72.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.72.0'>v1.72.0</a>.</em>

**Comment by crslade** [comment]
Any hints on how to update to this version on an Oracle ARM machine? It seems like it cannot find an ARM65/v8 image and defaults to an AMD image: Here is the error I get: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8).

**Comment by uninvitedguest** [comment]
I'm excited for hardware acceleration but not jumping on it just yet. How would trying to load the hwaccel.yml work when using a stack in Portainer? I'm not aware of how it could be "stored in the same directory" as the compose.yaml.

**Reply by jrasm91** [reply]
You can copy/paste the relevant section to effectively "inline it".

**Comment by sebastien247** [comment]
Why armv7 is no longer supported ? I wanted to install it on my NAS My Cloud EX2 Ultra.

**Reply by bo0tzz** [reply]
armv7 is an outdated architecture that is no longer supported by some of our dependencies and was causing issues with overly long build times.
